packages/next-server/server/api-utils.ts

```diff
@@ -0,0 +1,30 @@
+import { IncomingMessage, IncomingHttpHeaders } from 'http'
+import { parse as parseCookie } from 'cookie'
+import { parse } from 'url'
+
+/**
+ * Parse cookies from request header
+ * @param cookie from headers
+ */
+export function parseCookies({ cookie }: IncomingHttpHeaders) {
+  // If there is no cookie return empty object
+  if (!cookie) {
+    return {}
+  }
+
+  return parseCookie(cookie)
+}
+
+/**
+ * Parsing query arguments from request `url` string
+ * @param url of request
+ * @returns Object with key name of query argument and its value
+ */
+export function parseQuery({ url }: IncomingMessage) {
+  if (url) {
+    const { query } = parse(url, true)
```

Comment:
This api is deprecated, see here: https://nodejs.org/docs/latest/api/url.html#url_legacy_url_api